Paintroid/src/main/java/org/catrobat/paintroid/dialog/SaveInformationDialog.kt

```diff
@@ -0,0 +1,192 @@
+        onViewCreated(customLayout, savedInstanceState)
+        return AlertDialog.Builder(requireContext(), R.style.PocketPaintAlertDialog)
+            .setTitle(R.string.dialog_save_image_title).setView(customLayout)
+            .setPositiveButton(R.string.save_button_text) { dialog, which ->
+                FileIO.filename = imageName.text.toString()
+                if (permission != PERMISSION_EXTERNAL_STORAGE_SAVE_COPY && FileIO.checkIfDifferentFile(
+                        FileIO.getDefaultFileName()
+                    ) != IS_NO_FILE
+                ) {
+                    presenter.showOverwriteDialog(permission, isExport)
+                } else {
+                    presenter.switchBetweenVersions(permission, isExport)
+                }
+                dismiss()
+            }
+            .setNegativeButton(R.string.cancel_button_text) { dialog, which -> dismiss() }
+            .create()
+    }
+
+    private fun initializeViews(customLayout: View) {
+        specificFormatLayout =
+            customLayout.findViewById(R.id.pocketpaint_save_format_specific_options)
+        jpgView = inflater.inflate(
+            R.layout.dialog_pocketpaint_save_jpg_sub_dialog,
+            specificFormatLayout,
+            false
+        )
+        mySeekbar = jpgView.findViewById(R.id.pocketpaint_jpg_seekbar_save_info)
+        percentage = jpgView.findViewById(R.id.pocketpaint_percentage_save_info)
+        mySpinner = customLayout.findViewById(R.id.pocketpaint_save_dialog_spinner)
+        informationButton = customLayout.findViewById(R.id.pocketpaint_btn_save_info)
+        imageName = customLayout.findViewById(R.id.pocketpaint_image_name_save_text)
+    }
+
+    private fun initializeFunctioning() {
+        val spinnerArray = arrayListOf("png", "jpg", "ora")
+        val adapter =
+            ArrayAdapter(mySpinner.context, android.R.layout.simple_spinner_item, spinnerArray)
+        adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)
+        mySpinner.adapter = adapter
+        val percentageString = FileIO.compressQuality.toString()
+        percentage.text = percentageString
+        imageName.text = setName
+        mySeekbar.progress = FileIO.compressQuality
+        mySeekbar.setOnSeekBarChangeListener(this)
+        mySpinner.onItemSelectedListener = this
+        informationButton.setOnClickListener {
+            when {
+                FileIO.isCatrobatImage -> presenter.showOraInformationDialog()
+                FileIO.compressFormat == Bitmap.CompressFormat.JPEG -> presenter.showJpgInformationDialog()
+                else -> presenter.showPngInformationDialog()
+            }
+        }
+    }
+
+    private fun setFileDetails(
+        compressFormat: Bitmap.CompressFormat,
+        isCatrobatImage: Boolean,
+        ending: String
+    ) {
+        if (this::specificFormatLayout.isInitialized) {
+            specificFormatLayout.removeAllViews()
+        }
```

Comment:
```suggestion
        if (specificFormatLayout.isInitialized) {
            specificFormatLayout.removeAllViews()
        }
```
Ok, you won't get rid of this lateinit var ;)
But please be consistent with "this" ;D

Reply:
`this::` is required here😅

Reply:
`this::` no longer required